servicetalk-http-api/src/main/java/io/servicetalk/http/api/Http2SettingsBuilder.java

```diff
@@ -0,0 +1,229 @@
+                    MAX_FRAME_SIZE_UPPER_BOUND + "]");
+        }
+        settings.put(MAX_FRAME_SIZE, (long) value);
+        return this;
+    }
+
+    /**
+     * Set the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_MAX_HEADER_LIST_SIZE</a>.
+     *
+     * @param value The value.
+     * @return {@code this}.
+     */
+    public Http2SettingsBuilder maxHeaderListSize(long value) {
+        validate32Unsigned(value);
+        settings.put(MAX_HEADER_LIST_SIZE, value);
+        return this;
+    }
+
+    /**
+     * Build the {@link Map} that represents
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.1">HTTP/2 Setting</a>.
+     * @return the {@link Map} that represents
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.1">HTTP/2 Setting</a>.
+     */
+    public Http2Settings build() {
+        return new DefaultHttp2Settings(new HashMap<>(settings));
+    }
+
+    private static void validate32Unsigned(long value) {
+        if (value < 0 || value >= MAX_UNSIGNED_INT) {
+            throw new IllegalArgumentException("value: " + value + "(expected [0, " + MAX_UNSIGNED_INT + "]");
+        }
+    }
+
+    private static void validate31Unsigned(long value) {
+        if (value < 0 || value >= Integer.MAX_VALUE) {
+            throw new IllegalArgumentException("value: " + value + "(expected [0, " + Integer.MAX_VALUE + "]");
+        }
+    }
+
+    private static final class DefaultHttp2Settings implements Http2Settings {
+        private final Map<Character, Long> settings;
+
+        private DefaultHttp2Settings(final Map<Character, Long> settings) {
+            this.settings = settings;
+        }
+
+        @Nullable
+        @Override
+        public Long headerTableSize() {
+            return settings.get(HEADER_TABLE_SIZE);
+        }
+
+        @Nullable
+        @Override
+        public Long maxConcurrentStreams() {
+            return settings.get(MAX_CONCURRENT_STREAMS);
+        }
+
+        @Nullable
+        @Override
+        public Integer initialWindowSize() {
+            return toInteger(settings.get(INITIAL_WINDOW_SIZE));
+        }
+
+        @Nullable
+        @Override
+        public Integer maxFrameSize() {
+            return toInteger(settings.get(MAX_FRAME_SIZE));
+        }
+
+        @Nullable
+        @Override
+        public Long maxHeaderListSize() {
+            return settings.get(MAX_HEADER_LIST_SIZE);
+        }
+
+        @Nullable
+        @Override
+        public Long settingValue(final char identifier) {
+            return settings.get(identifier);
+        }
+
+        @Override
+        public void forEach(final BiConsumer<? super Character, ? super Long> action) {
+            settings.forEach(action);
+        }
+
+        @Override
+        public String toString() {
+            return settings.toString();
```

Comment:
The output of this impl is: `{=16384, =10}`.
Consider implementing it in the way it prints hex value for the identifier and maybe a string name for known identifiers.